### ISSUE TITLE
Hablity to name the whole ZCiCal ?

### ISSUE THREAD

**Issue by mans17** [phpfui/icalendar#9]
There is possibility to add a user friendly name to calendars.

Is this already possible (and I miss it in examples) or is it to be implemented ?

OUTPUT would look something like this, for better compatibulity (see : https://stackoverflow.com/questions/17152251/specifying-name-description-and-refresh-interval-in-ical-ics-format) :
```
NAME:My Calendar Name
X-WR-CALNAME:My Calendar Name
DESCRIPTION:A description of my calendar
X-WR-CALDESC:A description of my calendar
```

(great job for the library, works like a charm !)

**Comment by phpfui** [comment]
Glad you like the package. It is a very basic low level library, but that also allows you to add your own custom stuff. Apparently there is no real standard for ICalendar objects.  See the other open issues where I explain things (I am not the original author).

But I think you can do what you want by looking at [the simple example here](https://github.com/phpfui/icalendar/blob/master/examples/simpleevent.php).

You probably want to do something like this:
```php
// create the ical object
$icalobj = new \ICalendarOrg\ZCiCal();

// create the event within the ical object
$eventobj = new \ICalendarOrg\ZCiCalNode('VEVENT', $icalobj->curnode);

// add title
$eventobj->addNode(new \ICalendarOrg\ZCiCalDataNode('NAME:My Calendar Name'));
```

Let me know what you discover.  I will probably leave this issue open for documentation purposes, so your response will hopefully be useful to other users.

**Comment by mans17** [comment]
Thank you for your quick reply. My problem is that I want the line to be ABOVE the first "BEGIN:VEVENT".
=> The code you gave me gives something like 👍 
```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//ZContent.net//ZapCalLib 1.0//EN
CALSCALE:GREGORIAN
(...)
BEGIN:VEVENT
NAME:My Calendar Name
END:VEVENT
END:VCALENDAR
```

What I want (and obtained with this very dirty patch) is : 
```
BEGIN:VCALENDAR
VERSION:2.0
NAME:My Calendar Name
PRODID:-//ZContent.net//ZapCalLib 1.0//EN
CALSCALE:GREGORIAN
(...)
BEGIN:VEVENT
(...)
END:VEVENT
END:VCALENDAR
```

Here is my code to insert the lines, at third position of the ICS generated file (and the Name is recognized by google Calendar on imports)
```
        $data2 = $icalobj->export();
        $headers = "VERSION:2.0\r\nNAME:My Calendar Name\r\nX-WR-CALNAME:My Calendar Name\r\nDESCRIPTION:My Calendar Description\r\nX-WR-CALDESC:My Calendar Description\r\n";
        $data = str_replace ("VERSION:2.0\r\n", $headers, $data2);
```

I tried something like `$icalobj->addNode` but it did not seem to be accepted.

**Comment by phpfui** [comment]
This is a really awful hack, but better than what you are doing:

```php
// create the ical object
$icalobj = new \ICalendarOrg\ZCiCal();

// add name
$nameNode = new \ICalendarOrg\ZCiCalDataNode('NAME:My Calendar Name');
$insertPosition = array_search('PRODID', array_keys($icalobj->tree->data));
// Create a new array with the new element inserted
$icalobj->tree->data = array_slice($icalobj->tree->data, 0, $insertPosition) + [$nameNode->getName() => $nameNode] + array_slice($icalobj->tree->data, $insertPosition);

// continue adding stuff
```

What needs to happen is ZCiCal needs an addDataNode method with a optional after this data node parameter.

I will add that soon.

**Comment by phpfui** [comment]
Check out the V1.1.0 release.

**Comment by mans17** [comment]
I fixed my code with 1.1.0, and it works like a charm !

_TBH, my code contained a comment in french just above the line you seemed to love << ## please do not throw up when reading this : >>_ :-)

Thank you so much for your reactivity, and help cleaning up my mess :-)

**Comment by phpfui** [comment]
Good suggestion actually. Glad it works for you.

It is not my design, but it seems to work.
